distributed/utils_test.py

```diff
@@ -1750,9 +1750,17 @@ def check_thread_leak():
             # Raise an error with information about leaked threads
             from distributed import profile
 
-            bad_thread = bad_threads[0]
-            call_stacks = profile.call_stack(sys._current_frames()[bad_thread.ident])
-            assert False, (bad_thread, call_stacks)
+            lines: list[str] = [f"{len(bad_threads)} thread(s) were leaked from test\n"]
+            for i, thread in enumerate(bad_threads, 1):
+                lines.append(
+                    f"------ Call stack of leaked thread {i}/{len(bad_threads)}: {thread} ------"
+                )
+                lines.append(
+                    "".join(profile.call_stack(sys._current_frames()[thread.ident]))
```

Comment:
I think it's better to call sys._current_frames once and then delete it when finished